azure-pipelines.yml

```diff
@@ -11,34 +11,77 @@ pr:
       - develop
   drafts: false
 
-pool:
-  vmImage: ubuntu-latest
-
-steps:
-- task: UseDotNet@2
-  displayName: 'DotNet Install'
-  inputs:
-    packageType: 'sdk'
-    version: '3.x'
-
-- task: DotNetCoreCLI@2
-  displayName: 'Dotnet Build'
-  inputs:
-    command: 'build'
-    projects: $(projects)
-    arguments: '--output $(Build.ArtifactStagingDirectory) $(styleCopEnabled)'
+variables:
+  projects: '**/**/*.sln'
+
+stages:
+- stage: Build
+  displayName: Build
+
+  jobs: 
+  - job: Build
+    pool:
+      vmImage: ubuntu-latest
+
+    steps:
+    - task: UseDotNet@2
+      displayName: 'DotNet Install'
+      inputs:
+        packageType: 'sdk'
+        version: '3.x'
+
+    - task: DotNetCoreCLI@2
+      displayName: 'Dotnet Build'
+      inputs:
+        command: 'build'
+        projects: $(projects)
+        arguments: '--output $(Build.ArtifactStagingDirectory) $(styleCopEnabled)'
     
-- task: DotNetCoreCLI@2
-  displayName: 'Dotnet Test'
-  inputs:
-    command: 'test'
-    projects: $(testsProjects)
-
-- task: PublishBuildArtifacts@1
-  displayName: 'Dotnet Drop Artifacts'
-  condition: and(succeeded(), ne(variables['Build.Reason'], 'PullRequest'))
-  inputs:
-    PathtoPublish: '$(Build.ArtifactStagingDirectory)'
-    ArtifactName: 'Drop'
-    publishLocation: 'Container'
+    - task: DotNetCoreCLI@2
+      displayName: 'Dotnet Test'
+      inputs:
+        command: 'test'
+        projects: $(testsProjects)
 
+    - task: ArchiveFiles@2
+      displayName: 'Archive files'
+      inputs:
+        rootFolderOrFile: '$(Build.ArtifactStagingDirectory)'
+        includeRootFolder: false
+        archiveType: zip
+        archiveFile: $(Build.ArtifactStagingDirectory)/$(Build.BuildId).zip
+        replaceExistingArchive: true
+
+    - task: PublishBuildArtifacts@1
+      displayName: 'Drop Artifact'
+      condition: and(succeeded(), ne(variables['Build.Reason'], 'PullRequest'))
+      inputs:
+        PathtoPublish: '$(Build.ArtifactStagingDirectory)/$(Build.BuildId).zip'
+        ArtifactName: 'drop'
+        publishLocation: 'Container'
+
+- stage: Deploy
+  displayName: Deploy
+  dependsOn: Build
+  condition: and(succeeded(), eq(variables['Build.SourceBranch'], 'refs/heads/develop'))
+
+  jobs: 
+  - job: Deploy
+    displayName: Deploy
+    pool: 
+      vmImage: 'ubuntu-latest'
```

Comment:
let's remove pool here, we only need it at pipeline level https://docs.microsoft.com/en-us/azure/devops/pipelines/yaml-schema?view=azure-devops&tabs=schema%2Cparameter-schema#pool